scripts/blt/composer-version-convert.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * Usage:
+ * php -f convert-to-drupal-org-packagist.php $(pwd)/composer.json
+ */
+
+$composer_json_filepath = $argv[1];
+$composer_json = json_decode(file_get_contents($composer_json_filepath), TRUE);
+
+// Change drupal/lightning to acquia/lightning.
+if (isset($composer_json['require']['drupal/lightning'])) {
+  $composer_json['require']['acquia/lightning'] = $composer_json['require']['drupal/lightning'];
+  unset($composer_json['require']['drupal/lightning']);
+}
+
+// Update version constraints in require.
+foreach ($composer_json['require'] as $package_name => $version_constraint) {
+  if (strstr($package_name, 'drupal/')) {
+    $composer_json['require'][$package_name] = convert_version_constraint($version_constraint);
+  }
+}
+
+// Update version constraints in require-dev.
+foreach ($composer_json['require-dev'] as $package_name => $version_constraint) {
+  if (strstr($package_name, 'drupal/')) {
+    $composer_json['require-dev'][$package_name] = convert_version_constraint($version_constraint);
+  }
+}
+
+// Remove https://packagist.drupal-composer.org.
+foreach ($composer_json['repositories'] as $key => $repository) {
+  if ($repository['url'] == 'https://packagist.drupal-composer.org') {
+    unset($composer_json['repositories'][$key]);
+  }
+}
+
+// Add https://packages.drupal.org/8.
+$composer_json['repositories']['drupal'] = [
+  'type' => 'composer',
+  'url' => 'https://packages.drupal.org/8',
+];
+
+$output = json_encode($composer_json, JSON_PRETTY_PRINT|JSON_UNESCAPED_SLASHES);
+print $output;
+
+function convert_version_constraint($version_constraint) {
+
+  // 8.* => *
+  if (preg_match('/8\.\*$/', $version_constraint)) {
+    return "*";
+  }
+
+  // ~8 => *@stable
+  // ^8 => *@stable
+  if (preg_match('/^[\^~]8$/', $version_constraint)) {
+    return '*@stable';
+  }
+
+  // ~8.1.0-alpha1 > ~1.0.0-alpha1
+  // ^8.1.0-alpha1 > ^1.0.0-alpha1
+  // ^8.1.0 > ^1.0
+  // 8.1.0-alpha1 > 1.0.0-alpha1
+  // 8.1.0-beta12 > 1.0.0-beta12
+  // 8.12.0-rc22 > 12.0.0-rc22
+  if (preg_match('/^([\^~])?8(\.)?(\d+)?(\.\d+)?(-(alpha|beta|rc)\d+)?(\.\*)?(@(dev|alpha|beta|rc))?/', $version_constraint, $matches)) {
```

Comment:
Need to support more operators:
https://getcomposer.org/doc/articles/versions.md#range